django/db/models/expressions.py

```diff
@@ -485,12 +485,23 @@ def __init__(self, value, output_field=None):
         self.value = value
 
     def as_sql(self, compiler, connection):
-        if self.value is None:
+        val = self.value
+        if self._output_field_or_none is not None:
```

Comment:
There should be a comment why the check is against _output_field_or_none instead of output_field.